samples/storage/src/main/java/com/example/platform/storage/mediastore/MediaStoreQuery.kt

```diff
@@ -0,0 +1,169 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImage
+import com.example.platform.base.PermissionBox
+import com.google.android.catalog.framework.annotations.Sample
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+@Sample(
+    name = "MediaStore - Query",
+    description = "Query files indexed by MediaStore",
+    documentation = "https://developer.android.com/training/data-storage/shared/media#media_store",
+)
+@SuppressLint("MissingPermission")
+@Composable
+fun MediaStoreQueryScreen() {
+    val permission = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        READ_MEDIA_IMAGES
+    } else {
+        READ_EXTERNAL_STORAGE
+    }
+
+    PermissionBox(
+        permission = permission,
+        onGranted = { MediaStoreQueryContent() },
+        description = "Permission needed to access the device's photo library",
+    )
+}
+
+@RequiresPermission(anyOf = [READ_MEDIA_IMAGES, READ_EXTERNAL_STORAGE])
+@Composable
+fun MediaStoreQueryContent() {
+    val context = LocalContext.current
+    var files by remember { mutableStateOf(emptyList<FileEntry>()) }
+
+    LaunchedEffect(Unit) {
+        files = getImages(context.contentResolver)
+    }
+
+    Column(Modifier.fillMaxSize()) {
+        LazyColumn(Modifier.weight(1f)) {
+            item {
+                ListItem(
+                    headlineContent = {
+                        if (files.isNotEmpty()) {
+                            Text("${files.size} images found")
+                        }
+                    },
+                )
+                Divider()
+            }
+            items(files) { file ->
+                ListItem(
+                    leadingContent = {
+                        AsyncImage(
+                            model = file.uri,
+                            contentDescription = null,
+                            contentScale = ContentScale.Crop,
+                            modifier = Modifier
+                                .size(64.dp)
+                                .aspectRatio(1f),
+                        )
+                    },
+                    headlineContent = {
+                        Text(file.name, maxLines = 1, overflow = TextOverflow.Ellipsis)
+                    },
+                    supportingContent = { Text(file.mimeType) },
+                    trailingContent = { Text(Formatter.formatShortFileSize(context, file.size)) },
+                )
+                Divider()
+            }
+        }
+    }
+}
+
+suspend fun getImages(contentResolver: ContentResolver): List<FileEntry> {
```

Comment:
Since this is the main part of the sample. Could you add some javadocs explaining the function a bit?

Also, make it private